lib/cisco_node_utils/snmp_notification_receiver.rb

```diff
@@ -0,0 +1,158 @@
+# November 2015, Jonathan Tripathy
+#
+# Copyright (c) 2014-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative 'node_util'
+
+module Cisco
+  # SnmpNotificationReceiver - node utility class for SNMP server management
+  class SnmpNotificationReceiver < NodeUtil
+    attr_reader :name
+
+    def initialize(name,
+                   instantiate=true,
+                   type='',
+                   version='',
+                   security='',
+                   username='',
+                   port='',
+                   vrf='',
+                   source_interface='')
+
+      fail TypeError unless name.is_a?(String)
+      @name = name
+
+      fail TypeError unless type.is_a?(String)
+
+      fail TypeError unless version.is_a?(String)
+
+      fail TypeError unless security.is_a?(String)
+
+      fail TypeError unless username.is_a?(String)
+
+      fail TypeError unless port.is_a?(String)
```

Comment:
Everything is a string in the Cisco CLI. I can port to a Integer though if you would like. It just means I would have to cast in the Puppet provider and create a separate munge method for integers.